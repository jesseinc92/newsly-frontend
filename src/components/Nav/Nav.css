.Nav {
  text-align: left;
  width: 100vw;
  position: fixed;
  top: 0;
  left: 0;
  background-color: white;

  display: flex;
  justify-content: center;

  z-index: 1;
}

.Nav-burger-wrapper,
.Nav-profile-wrapper,
.Nav-close {
  height: 5rem;
  width: 5rem;
  font-size: 2rem;
  display: flex;
  align-items: center;
  justify-content: center;
}

.Nav-burger-wrapper {
  margin-right: auto;
}

.Nav-close-wrapper {
  position: fixed;
  top: 0;
  left: 0;
  height: 0%;
  width: 100vw;
  overflow: hidden;
  background-color: white;
}

.Nav-burger div {
  background-color: black;
  width: 2rem;
  height: 3px;
  margin: 0.5rem;
  border-radius: 1.5px;
}

.Nav-burger div:nth-child(2) {
  width: 1.5rem;
}

.Nav-burger div:last-child {
  width: 1.75rem;
}

.Nav-links {
  display: flex;
  flex-direction: column;
  font-family: 'Fjalla One', sans-serif;
  font-size: 2.5rem;
  margin-top: 15vh;
}

.Nav-links a {
  margin-top: 1.25rem;
  margin-left: 2rem;
}

/* Tablet sizing */
@media screen and (min-width: 768px) {
  
  .Nav-links {
    font-size: 4rem;
  }

}

/* Laptop/Desktop sizing */
@media screen and (min-width: 1024px) {

  .Nav-burger-wrapper,
  .Nav-close,
  .Nav-profile-wrapper {
    display: none;
  }

  .Nav-close-wrapper {
    height: 5vw;
    width: 100vw;

    display: flex;
    justify-content: center;
    align-items: center;
  }

  .Nav-links {
    width: 100vw;
    margin-top: unset;
    font-size: 1.5vw;

    flex-direction: row;
    align-items: center;
  }

  .Nav-links a {
    margin: 0 2rem;
  }

  .Nav-links a:nth-child(2) {
    margin-right: auto;
  }

}