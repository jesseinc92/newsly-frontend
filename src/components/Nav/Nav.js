import React, { useContext } from 'react';
import { NavLink } from 'react-router-dom';
import UserContext from '../../UserContext';
import Button from '../Button/Button';

const Nav = () => {
  const { user, handleLogout } = useContext(UserContext);

  return (
    <nav className='Nav'>
      <NavLink to='/'>Home</NavLink>
      <NavLink to='/front-page'>Front Page</NavLink>
      <NavLink to='/search'>Search</NavLink> 
      {user ?
      <>
        <NavLink to={`/user/${user.username}`}>Hello, {user.firstName}</NavLink>
        <NavLink to={`/user/${user.username}/bookmarks`}>Bookmarks</NavLink>
        <Button text='Logout' handler={handleLogout} />
      </>
      :
      <>
        <NavLink to='/user/login'>Login</NavLink>
        <NavLink to='/user/register'>Register</NavLink>
      </>
      }
    </nav>
  )
}

export default Nav;