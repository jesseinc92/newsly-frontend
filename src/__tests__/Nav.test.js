import React from 'react';
import { screen } from '@testing-library/react';
import renderWithContext from '../helpers/testContext';
import Nav from '../components/Nav/Nav';
import { MemoryRouter } from 'react-router-dom';

it('renders the component', function() {
  renderWithContext(<MemoryRouter>
                      <Nav />
                    </MemoryRouter>);
});

it('matches the snapshot', function() {
  const { asFragment } = renderWithContext(<MemoryRouter>
                                            <Nav />
                                          </MemoryRouter>);
  expect(asFragment()).toMatchSnapshot();
});

it('displays the first name if logged in', function() {
  renderWithContext(<MemoryRouter><Nav /></MemoryRouter>);
  const link = screen.getByText('Hello, test');
  expect(link).toBeInTheDocument();
});

it('displays the login button if not logged in', function() {
  renderWithContext(<MemoryRouter><Nav /></MemoryRouter>, null);
  const link = screen.getByText('Login');
  expect(link).toBeInTheDocument();
});