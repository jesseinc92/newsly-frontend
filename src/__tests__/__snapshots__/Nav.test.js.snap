// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<DocumentFragment>
  <nav
    class="Nav"
  >
    <div
      class="Nav-burger-wrapper"
    >
      <div
        class="Nav-burger"
      >
        <div />
        <div />
        <div />
      </div>
    </div>
    <div
      class="Nav-profile-wrapper"
    >
      <a
        href="/user/test1"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-user "
          data-icon="user"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M224 256c70.7 0 128-57.31 128-128s-57.3-128-128-128C153.3 0 96 57.31 96 128S153.3 256 224 256zM274.7 304H173.3C77.61 304 0 381.6 0 477.3c0 19.14 15.52 34.67 34.66 34.67h378.7C432.5 512 448 496.5 448 477.3C448 381.6 370.4 304 274.7 304z"
            fill="currentColor"
          />
        </svg>
      </a>
    </div>
    <div
      class="Nav-close-wrapper"
    >
      <div
        class="Nav-close"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-xmark "
          data-icon="xmark"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 320 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M310.6 361.4c12.5 12.5 12.5 32.75 0 45.25C304.4 412.9 296.2 416 288 416s-16.38-3.125-22.62-9.375L160 301.3L54.63 406.6C48.38 412.9 40.19 416 32 416S15.63 412.9 9.375 406.6c-12.5-12.5-12.5-32.75 0-45.25l105.4-105.4L9.375 150.6c-12.5-12.5-12.5-32.75 0-45.25s32.75-12.5 45.25 0L160 210.8l105.4-105.4c12.5-12.5 32.75-12.5 45.25 0s12.5 32.75 0 45.25l-105.4 105.4L310.6 361.4z"
            fill="currentColor"
          />
        </svg>
      </div>
      <div
        class="Nav-links"
      >
        <a
          aria-current="page"
          class="active"
          href="/"
        >
          Home
        </a>
        <a
          class=""
          href="/front-page"
        >
          Front Page
        </a>
        <a
          class=""
          href="/search"
        >
          Search
        </a>
        <a
          class=""
          href="/user/test1"
        >
          Hello, test
        </a>
        <a
          class=""
          href="/user/test1/bookmarks"
        >
          Bookmarks
        </a>
        <a
          href="/"
        >
          Logout
        </a>
      </div>
    </div>
  </nav>
</DocumentFragment>
`;
